### ISSUE TITLE
[Refactor] tests for backend

### ISSUE THREAD

**Issue by mthrok** [pytorch/audio#1230]
The number of unit tests we have in torchaudio has grown a lot. It is critical to keep tests organized so as to make it easy for contributors to write tests for the feature they are working on.

There are currently following tests for backend. We would like to consolidate them.
 - [test/torchaudio_unittest/backend_test.py](https://github.com/pytorch/audio/blob/3651412bbcb0c4faf56bc7be7fe5e5ae2a7a63fd/test/torchaudio_unittest/backend_test.py)
 - [test/torchaudio_unittest/io_test.py](https://github.com/pytorch/audio/blob/3651412bbcb0c4faf56bc7be7fe5e5ae2a7a63fd/test/torchaudio_unittest/io_test.py)
 - [test/torchaudio_unittest/sox_io_backend/*_test.py](https://github.com/pytorch/audio/blob/3651412bbcb0c4faf56bc7be7fe5e5ae2a7a63fd/test/torchaudio_unittest/sox_io_backend)
 - [test/torchaudio_unittest/soundfile_backend/*_test.py](https://github.com/pytorch/audio/blob/3651412bbcb0c4faf56bc7be7fe5e5ae2a7a63fd/test/torchaudio_unittest/soundfile_backend)

## Steps
1. Add `backend` directory in `test/torchaudio_unittest` directory.
2. Move the tests mentioned above in `test/torchaudio_unittest/backend` directory with the following rename
    - `backend_test.py` -> `utils_test.py`
    - `io_test.py` -> `legacy_test.py`
    - `sox_io_backend` directory -> `sox_io` directory (The names of the individual scripts in the directory should stay same)
    - `soundfile_backend` directory -> `soundfile` directory (The names of the individual scripts in the directory should stay same)
3. Fix the `import` statements.
4. Resolve [this TODO](https://github.com/pytorch/audio/blob/3651412bbcb0c4faf56bc7be7fe5e5ae2a7a63fd/test/torchaudio_unittest/soundfile_backend/info_test.py#L16-L17).
    - Create `common.py` under `test/torchaudio_unittest/backend` directory
    - Move the functions there
    - Update the import statements.

## Building and Testing

1. Install nightly build of PyTorch
2. Clone repo and build torchaudio
    ```
    # mac
    BUILD_SOX=1 MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py develop
    # linux
    BUILD_SOX=1 python setup.py clean develop
    ```
3. Set `PATH` variable so that the `sox` command created as a part of torchaudio build process is used.
    `export PATH="<path_to_torchaudio>/third_party/install/bin:${PATH}"`
4. Run test
    Run test with `pytest test/torchaudio_unittest/backend -v`

**Comment by prabhat00155** [comment]
I can work on this.

**Comment by mthrok** [comment]
closed via #1239

**Comment by mthrok** [comment]
Thanks!
